### ISSUE TITLE
Could you guys let some of the team members go to update WCF of Xamarin.Android and Xamarin.iOS?

### ISSUE THREAD

**Issue by huangjinshe** [dotnet/wcf#1688]
Since WCF supported cross platform, We All need some features on the other platform such like Duplex Channel. But until now, it's not finished for Xamarin.Android and Xamarin.iOS. 

We know Xamarin already belong to the Microsoft now, It would be very helpful if WCF team just let some one go to improve the feature of WCF for Xamarin, that will be very helpful for every one.

There are a lot of user want it too: [More advanced/complete support for WCF](https://xamarin.uservoice.com/forums/144858-xamarin-platform-suggestions/suggestions/2837262-more-advanced-complete-support-for-wcf)

**Comment by zhenlan** [comment]
@terrajobst can you help to route the feedback to Xamarin folks? It will be great if Xamarin can leverage WCF code on Core or full framework.

**Comment by huangjinshe** [comment]
@terrajobst ,sorry for bothering, but is there any response from Xamarin team?  Because if that will take too much time such like a year. I think we need to replace another solution instead of WCF for other platform, currently still waiting for that.

**Comment by terrajobst** [comment]
Hey @marek-safar and @akoeplinger. Could you chime in?

**Comment by akoeplinger** [comment]
@marek-safar as far as I know Marcos did a prototype on using the corefx WCF sources, do you know where we ended up with this?

**Comment by marek-safar** [comment]
I am not sure why do you think WCF is cross platform but Xamarin WCF support is limited. I don't think this is going to change anytime soon (if ever) as WCF is large and has many hard to port dependencies.

We plan to (would like to) support full subset of WCF .NET Core client APIs but that work has been postponed few times already.

**Comment by huangjinshe** [comment]
@marek-safar ,I don't know about other features, but only the Duplex channel we want it so badly for so long time. I really hope you guys has some priority levels for the new implementation.

**Comment by huangjinshe** [comment]
@marek-safar , Is there any good news now? espetially for TCP Duplex channel.

**Comment by EdHyman** [comment]
The duplexchannelfactory was working as intended on a previous version of Xamarin Studio.  I upgrade to a new version of OS (from Yosemite to El Capitan) and install Xamarin updates as well, then I receive this nonimplementedexception error.  My previously compiled code running on a Mac laptop (old Yosemite OS) executes without error.  Has anyone put together a work-around, for instance two one way channels?
Ed H  (edhyman@icloud.com)

**Comment by ivanbottero** [comment]
Is there any news for TCP Duplex channel?

**Comment by marek-safar** [comment]
No update for now but I'll see if we can schedule it anytime soon

**Comment by ArgusMagnus** [comment]
I would highly appreciate tcp duplex channel support

**Comment by huangjinshe** [comment]
@marek-safar Please schedule it soon.

**Comment by ArgusMagnus** [comment]
Any progress on this?

**Comment by marek-safar** [comment]
This is large effort for us with many implications to resolve. Unfortunately, I don't see us leveraging dotnet WCF code anytime soon.

**Comment by huangjinshe** [comment]
@marek-safar  I thought you guys will do that few months ago. You sounds like you guys giving up that now.....

**Comment by kartalbas** [comment]
so we have to wait until NET Core replaces mono?

**Comment by ArgusMagnus** [comment]
@marek-safar Is it save to assume, that tcp duplex support is not coming to Xamarin.Android by the end of this year?
Is there a realistic chance that this will happen in the future at all?

**Comment by marek-safar** [comment]
@ArgusMagnus yes, that's the safe to assume but it's still in the plan for the future

**Comment by zhenlan** [comment]
@marek-safar what's the repo for Xamarin/mono so we can move this issue there for better tracking since this repo is only for WCF on .NET Core?

**Comment by marek-safar** [comment]
@zhenlan best to report the issue at https://bugzilla.xamarin.com/

**Comment by huangjinshe** [comment]
@marek-safar , @zhenlan Don't need to do that, because I already did that almost 1 year or maybe 1.5 year ago......Turns out there is no one responsible for that.

**Comment by huangjinshe** [comment]
@marek-safar , @zhenlan 
[Request add WCF Duplex Channel support for Mono(for Xamarin.Android and Xamarin.IOS)](https://bugzilla.xamarin.com/show_bug.cgi?id=41392)
[DuplexChannelFactory.CreateChannel throw exception NotImplementedException](https://bugzilla.xamarin.com/show_bug.cgi?id=44513)

Maybe there are more, but the result is.....none.

**Comment by huangjinshe** [comment]
@marek-safar , Hi, Is there any news for this? It take too much time. If UWP dead right now, we need to use Android or iOS version of WCF feature more soon.

**Comment by marek-safar** [comment]
@huangjinshe no update at this point

**Comment by huangjinshe** [comment]
Is there any good news.....I almost lose my mind......

**Comment by huangjinshe** [comment]
Is there any news in this year?  Microsoft is trying to support more and more, they want to be a service, the reason we use WCF because we want some advanced feature such like TCD Duplex Channel or like certification encryption, why stop updating it after finish some simple feature? , If we just use simple HTTP connection with json, then what's the point of use WCF? I hope WCF team bring these advanced feature to IOS and Android, let us use, not only stop after bring it after dead Windows phone !

**Comment by terrajobst** [comment]
@huangjinshe 

At Build we've announced our plans for .NET Core 3.0 which is focused on bringing the client tech (WinForms/WPF) to .NET Core (but it will remain Windows-only). Based on customer feedback we've prioritized this work over porting more server technologies. That doesn't mean those aren't gonna happen but it does mean they are currently taking the backseat. Please keep the feedback coming so we can prioritize it accordingly moving forward!

**Comment by huangjinshe** [comment]
@terrajobst 

Thank you, that's a really good news, But still only for Windows, currently we need really need much more support for Android and iOS. I just hope you guys just use .NET Core replace Xamarin.iOS and Xamarin.Android (currently those based on mono), because .NET core growing speed too much faster than mono, mono has too much limits, No advanced feature.

Maybe that's much more better when use  Xamarin/.NET Core, it should be better than  Xamarin/mono.

**Comment by terrajobst** [comment]
.NET Core will remain a cross-platform and we're working with the Mono/Xamarin teams to share more code to get a a more converged library stack. .NET has always been about also giving you access to native OS features and we'll continue to do so. However, I do agree with you that we need to also push the envelope how far you can go with cross-platform application and UI definition, but that's a next step.
